R/slide.R

```diff
@@ -111,8 +121,23 @@ epi_slide <- function(
     )
   }
 
-  # Function body starts
+  # Validate arguments
   assert_class(.x, "epi_df")
+  if (checkmate::test_class(.x, "grouped_df")) {
+    expected_group_keys <- .x %>%
+      key_colnames() %>%
+      kill_time_value() %>%
+      sort()
+    if (.x %>% groups() %>% as.character() %>% sort() != expected_group_keys) {
+      cli_abort(
+        "epi_slide: `.x` must be either ungrouped or grouped by {expected_group_keys}. You may need to aggregate
+        your data first, see aggregate_epi_df().",
+        class = "epi_slide__invalid_grouping"
```

Comment:
```suggestion
        class = "epiprocess__epi_slide__invalid_grouping"
```
iirc rlang/tidyverse suggests including package name in custom error classes